src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -1718,4 +1718,46 @@ abstract class MergeIntoSuiteBase
       }
     }
   }
+
+  protected def testMatchedOnlyOptimization(
+                                             name: String)(
+                                             source: Seq[(Int, Int)],
+                                             target: Seq[(Int, Int)],
+                                             mergeOn: String,
+                                             mergeClauses: MergeClause*) (
+                                             result: Seq[(Int, Int)]): Unit = {
+    Seq(true, false).foreach { matchedOnlyEnabled =>
+      withSQLConf(DeltaSQLConf.MERGE_MATCHED_ONLY_ENABLED.key -> matchedOnlyEnabled.toString) {
+        testExtendedMerge(
+          s"matched only clause enabled = ${matchedOnlyEnabled} - $name") (
```

Comment:
please follow the test name structure of  insert-only-merge tests, that is, "insert only merge - ...."

Reply:
I don't understand this comment.  the way it is right now the name of the test will indicate whether the optimization is on or off and have detail about specific test like update or delete etc.  Seems the same as insert only except that my tests are run with and without the optimization so it's useful to indicate that in the test name....